### ISSUE TITLE
mAP code

### ISSUE THREAD

**Issue by sudhakar-eyeem** [pierluigiferrari/ssd_keras#4]
I have written a mAP code using your code. Should I add PR? 
Also, I am training using our own classifier model as base model. However, my model loss goes down only till 10-12 Epochs for pascal VOC. Do you think that 10-12 epochs is sufficient for Pascal voc, especially since my base model is very light weight not as heavy as VGG16.

**Comment by pierluigiferrari** [comment]
Yes, that would be great! I'm a bit swamped and might need a while to check it out, but it would be much appreciated!
Regarding your training loss: How many training iterations make up one of your epochs? What does your base model look like? I haven't done extensive testing on this yet, but the smaller the model, the less complexity it can learn, so it might be a possibility that your model relatively quickly reaches a point where it is too small to get much better. Of course that's a wild guess at this point without knowing what your base model is. Have you checked out a few example predictions to see what the results look like after 10-12 epochs?

**Comment by sudhakar-eyeem** [comment]
thank you for your answers. Well, I am using a batch size of 32 and using pascal voc dataset, total 360 iterations per epoch. My validation loss start getting saturation after 13-14 iterations and it changes once or twice till 30 iterations. 
I have checked the output and also evaluated mAP which is around 61% on pascal voc 2007 test data (reduction of mAP may be because of the base model that I am using, I am planning to change it and see if the problem is with base model)

**Comment by pierluigiferrari** [comment]
Ok good, 61% mAP in Pascal VOC is not phenomenal, but it's not bad either, so this result reduces the chance that there's a gross bug in the code somewhere. It would be interesting to know what the configuration of your base model is. One possibility is that your base model could indeed be too small to improve much beyond this value.
